<!-- 
  Launch height mapping
  Parameters:
  - Use debug_mode:=true to publish debug information
-->

<launch>
  <arg name="debug_mode" default="false" />

  <!-- Mapping node -->
  <node pkg="height_mapping" type="height_mapping_node" name="height_mapping_local" output="screen">
    <rosparam command="load" subst_value="true" file="$(find height_mapping)/config/height_mapping_node.yaml" />
    <param name="node/debug_mode" value="$(arg debug_mode)" />
  </node>

  <!-- Use grid_map_visualization package for map visualization -->
  <node pkg="grid_map_visualization" type="grid_map_visualization" name="height_mapping_vis" output="log">
    <rosparam command="load" file="$(find height_mapping)/config/include/heightmap_visualization.yaml" />
  </node>

</launch>